windows/devices/uwb/UwbDeviceConnector.cxx

```diff
@@ -395,46 +395,45 @@ UwbDeviceConnector::SetApplicationConfigurationParameters(uint32_t sessionId, st
-    //     auto uwbStatus = UwbCxDdi::To(uwbSetAppConfigParamsStatus.status);
-    //     if (!IsUwbStatusOk(uwbStatus)) {
-    //         resultPromise.set_exception(std::make_exception_ptr(UwbException(std::move(uwbStatus))));
-    //     } else {
-    //         // TODO: Could move this logic into its own UwbCxDdi::To() function
-    //         std::vector<std::tuple<::uwb::protocol::fira::UwbApplicationConfigurationParameterType, ::uwb::protocol::fira::UwbStatus>> appConfigParamsStatusList;
-    //         for (auto i = 0; i < uwbSetAppConfigParamsStatus.appConfigParamsCount; i++)
-    //         {
-    //             auto paramType = UwbCxDdi::To(uwbSetAppConfigParamsStatus.appConfigParamsStatus[i].paramType);
-    //             auto paramStatus = UwbCxDdi::To(uwbSetAppConfigParamsStatus.appConfigParamsStatus[i].status);
-
-    //             auto appConfigParamStatus = std::make_tuple(paramType, paramStatus);
-    //             appConfigParamsStatusList.push_back(appConfigParamStatus);
-    //         }
-    //         auto result = std::make_tuple(uwbStatus, std::move(appConfigParamsStatusList));
-    //         resultPromise.set_value(std::move(result));
-    //     }
-    // }
+
+    wil::shared_hfile handleDriver;
+    auto hr = OpenDriverHandle(handleDriver, m_deviceName.c_str());
+    if (FAILED(hr)) {
+        PLOG_ERROR << "failed to obtain driver handle for " << m_deviceName << ", hr=" << std::showbase << std::hex << hr;
+        resultPromise.set_exception(std::make_exception_ptr(UwbException(UwbStatusGeneric::Rejected)));
+        return resultFuture;
+    }
+
+    auto paramsBuffer = UwbCxDdi::From(applicationConfigurationParameters).data();
+    auto statusSize = offsetof(UWB_SET_APP_CONFIG_PARAMS_STATUS, appConfigParamsStatus[applicationConfigurationParameters.size()]);
+    std::vector<uint8_t> statusBuffer(statusSize);
+    BOOL ioResult = DeviceIoControl(handleDriver.get(), IOCTL_UWB_SET_APP_CONFIG_PARAMS, std::data(paramsBuffer), std::size(paramsBuffer), std::data(statusBuffer), statusSize, nullptr, nullptr);
+    if (!LOG_IF_WIN32_BOOL_FALSE(ioResult)) {
+        HRESULT hr = HRESULT_FROM_WIN32(GetLastError());
+        PLOG_ERROR << "error when sending IOCTL_UWB_SET_APP_CONFIG_PARAMS with sessionId " << sessionId << ", hr = " << std::showbase << std::hex << hr;
+        resultPromise.set_exception(std::make_exception_ptr(UwbException(UwbStatusGeneric::Rejected)));
+        return resultFuture;
+    } else {
+        PLOG_DEBUG << "IOCTL_UWB_SET_APP_CONFIG_PARAMS succeeded";
+        auto& uwbSetAppConfigParamsStatus = *reinterpret_cast<UWB_SET_APP_CONFIG_PARAMS_STATUS*>(std::data(statusBuffer));
+        auto uwbStatus = UwbCxDdi::To(uwbSetAppConfigParamsStatus.status);
+        if (!IsUwbStatusOk(uwbStatus)) {
+            resultPromise.set_exception(std::make_exception_ptr(UwbException(std::move(uwbStatus))));
+        } else {
+            // TODO: Could move this logic into its own UwbCxDdi::To() function

```

Comment:
FTFY 😄.

```suggestion
            // TODO: Should move this logic into its own UwbCxDdi::To() function
```

Since this is converting a DDI type, we definitely need both a `From()` and `To()` function for it, since the simulator driver will use them on both sides.